tidy3d/components/data/monitor_data.py

```diff
@@ -945,6 +946,113 @@ def _reorder_modes(
 
         return self.copy(update=update_dict)
 
+    def dot(self, field_data: ElectromagneticFieldData, conjugate: bool = True) -> xr.DataArray:
+        """Dot product (modal overlap) with another :class:`.FieldData` object. Both datasets have
+        to be frequency-domain data associated with a 2D monitor. Along the tangential directions,
+        the datasets have to have the same discretization. Along the normal direction, the monitor
+        position may differ and is ignored. Other coordinates (``frequency``, ``mode_index``) have
+        to be either identical or broadcastable. Broadcasting is also supported in the case in
+        which the other ``field_data`` has a dimension of size ``1`` whose coordinate is not in the
+        list of coordinates in the ``self`` dataset along the corresponding dimension. In that case,
+        the coordinates of the ``self`` dataset are used in the output.
+
+        Parameters
+        ----------
+        field_data : :class:`ElectromagneticFieldData`
+            A data instance to compute the dot product with.
+        conjugate : bool, optional
+            If ``True`` (default), the dot product is defined as ``1 / 4`` times the integral of
+            ``E_self* x H_other - H_self* x E_other``, where ``x`` is the cross product and ``*`` is
+            complex conjugation. If ``False``, the complex conjugation is skipped.
+
+        Note
+        ----
+            The dot product with and without conjugation is equivalent (up to a phase) for
+            modes in lossless waveguides but differs for modes in lossy materials. In that case,
+            the conjugated dot product can be interpreted as the fraction of the power of the first
+            mode carried by the second, but modes are not orthogonal with respect to that product
+            and the sum of carried power fractions exceed 1. In the non-conjugated definition,
+            orthogonal modes can be defined, but the interpretation of modal overlap as power
+            carried by a given mode is no longer valid.
+        """
+
+        # Flip integration direction for backward monitors
+        if self.monitor.direction == "-":
```

Comment:
nitpicky but this seems a bit cleaner to avoid repetition.
```py
sign = -1 if self.monitor.direction == "-" else 1.0
return sign * super().dot(field_data, conjugate)
```